nowcasting_dataset/time.py

```diff
@@ -296,3 +296,23 @@ def make_random_time_vectors(
         "time_30": utils.to_numpy(time_30.T).astype(np.int32),
         "time_60": utils.to_numpy(time_60.T).astype(np.int32),
     }
+
+
+def floor_minutes_dt(dt, minutes: Optional[int] = 30):
```

Comment:
oooh, the little I know about writing functions to process datetimes suggests that there are enough fiddly edge-cases to mean that it's almost always better to use existing, well-tested libraries for processing datetimes, like Pandas! :slightly_smiling_face: . So I'd be very tempted to replace this custom function with `return pd.Timestamp(dt).floor(pd.Timedelta(minutes=minutes))`

Reply:
I agree, just sometimes feels like an over kill to use pandas when we are in 'datetime' world. But yea, I think yoru right in this case